src/main/resources/META-INF/jqassistant-rules/cdi/cdi.xml

```diff
@@ -3,420 +3,255 @@
-              (:Type)-[:DECLARES]->(member),
-              (member)-[:OF_TYPE|RETURNS]->(type),
-              (member)-[:ANNOTATED_BY]->()-[:OF_TYPE]->(producesType:Type)
-            WHERE
-              (member:Field or member:Method)
-              and producesType.fqn="javax.enterprise.inject.Produces"
-            SET
-              type:JEE:Injectable,
-              member:JEE:BeanProducer
-            MERGE
-              (member)-[:PRODUCES]->(type)
+              (member:JEE:BeanProducer)
             RETURN
               member AS Producer
         ]]></cypher>
     </concept>
 
     <concept id="cdi:Disposes">
-        <description>Creates a relation DISPOSES between a parameter and its type if the parameter is annotated by @javax.enterprise.inject.Disposes.
+        <description>Returns all disposer methods.
         </description>
         <cypher><![CDATA[
             MATCH
-              (:Type)-[:DECLARES]->(disposeMethod:Method)-[:HAS]->(parameter:Parameter),
-              (parameter)-[:ANNOTATED_BY]->()-[:OF_TYPE]->(disposesType:Type),
-              (parameter)-[:OF_TYPE]->(type)
-            WHERE
-              disposesType.fqn="javax.enterprise.inject.Disposes"
-            MERGE
-              (parameter)-[:DISPOSES]->(type)
+              (:Type)-[:DECLARES]->(disposeMethod:Method)-[:HAS]->(parameter:Parameter)-[:DISPOSES]->(:Type)
             RETURN
               disposeMethod AS DisposeMethod
         ]]></cypher>
     </concept>
 
     <concept id="cdi:Qualifier">
-        <description>Labels all annotation types annotated by @javax.inject.Qualifier with "CDI" and "Qualifier" and adds the labels "CDI"
-            and "Nonbinding" to
-            all non-binding annotation values (i.e. which are annotated by @javax.enterprise.util.Nonbinding).
+        <description>Abstract concept that returns all annotation types labeled with "CDI" and "Qualifier".
         </description>
         <cypher><![CDATA[
             MATCH
-              (qualifier:Type)-[:ANNOTATED_BY]->()-[:OF_TYPE]->(qualifierType:Type)
-            WHERE
-              qualifierType.fqn = "javax.inject.Qualifier"
-            SET
-              qualifier:CDI:Qualifier
-            WITH
-              qualifier
-            MATCH
-              (qualifier)-[:DECLARES]->(attribute:Method),
-              (attribute)-[:ANNOTATED_BY]->()-[:OF_TYPE]->(nonbindingType:Type)
-            WHERE
-              nonbindingType.fqn = "javax.enterprise.util.Nonbinding"
-            SET
-              attribute:CDI:Nonbinding
+              (qualifier:Type:CDI:Qualifier)
             RETURN
               distinct qualifier AS Qualifier
         ]]></cypher>
     </concept>
 
     <concept id="cdi:Dependent">
-        <description>Labels all beans, fields or methods annotated by @javax.enterprise.context.Dependent with "CDI" and "Dependent".
+        <description>Abstract concept that returns all beans, fields or methods labeled with "CDI" and "Dependent".
```

Comment:
Ich fände es als Idee für jQA sinnvoll, Konzepte direkt als abstract="true" markieren zu können. Um sicherzustellen, dass in diesen dann der Graph nicht manipuliert wird, könnte die Abfrage in einer read-only Transaktion laufen.